src/Chapter1.hs

```diff
@@ -187,463 +184,460 @@ ghci> :q
+-- supposed to. Try evaluating the following expressions to see errors:
+--
+-- ghci> not 'a'
+-- ghci> max True 'x'
+-- ghci> 10 + True
+--
+-- This is a gentle way to get familiar with various error messages in Haskell.
+-- In some cases, the error messages could be challenging to decipher and
+-- understand their meaning. Haskell has a bad reputation for having not-so-helpful
+-- error messages in some situations. But, of course, such a small challenge won't
+-- stop you, right? You're a brave warrior, and you can finish all tasks despite
+-- all obstacles! And we are always here to help and to decrypt these ancient
+-- scripts together.
+
+-- |
+-- =🛡= Defining a function
+--
+-- We have already learned how to use different functions and operators in Haskell.
+-- Let's now check how they are defined and whether we can introduce our own.
+--
+-- When defining a function in Haskell, you write its type signature on the first
+-- line, and then its body on the following line(s). The type signature should be
+-- written immediately from the start of a line. Haskell is __indentation-__ and
+-- __layout-sensitive__ language, so this is important to keep in mind.
+--
+-- For example, here is the type signature of a function that takes a 'Double' and
+-- an 'Int', and then returns an 'Int':
+--
+-- @
+-- roundSubtract :: Double -> Int -> Int
+-- @
+--
+-- We have already seen the "::" sequence of symbols when practising our skills in
+-- GHCi. Now you know that this is the syntax for specifying types in your code as
+-- well.
+--
+-- The following line should be the function definition start line. You write the
+-- function name again and give argument names in the same order as you wrote types
+-- followed by the "=" sign. And you provide the function implementation after "=".
+--
+-- @
+-- roundSubtract x y = ceiling x - y
+-- @
+--
+-- ^ Here x corresponds to the 'Double', and y to 'Int'.
+--
+-- The body of the function can be as big as you want. However, don't forget about
+-- the indentation rules when your body exceeds the definition line.
+--
+-- The same function body can be written on a separate line, minding the
+-- indentation.
+--
+-- @
+-- roundSubtract x y =
+--     ceiling x - y
+-- @
+--
+-- Putting everything together, the complete function definition looks like this:
+--
+-- @
+-- roundSubtract :: Double -> Int -> Int
+-- roundSubtract x y = ceiling x - y
+-- @
+--
+-- Now you are ready for defining your own functions!
+
+-- |
+-- In our training, for some functions types are provided for you. For others, you
+-- need to write types manually to challenge yourself.
+--
+-- Don't forget the main rule:
+-- **Always provide type signatures for top-level functions in Haskell.**
+
+-- |
+-- =⚔️= Task 3
+--
+-- Below you see the function that finds a square of the sum of two integers. Your
+-- task is to specify the type of this function.
+--
+-- >>> squareSum 3 4
+-- 49
+squareSum :: Num a => a -> a -> a
```

Comment:
We explain `Num` in the following chapters, so you are one step ahead 😸

In here `Int -> Int -> Int` would also work, as a more specific type 🙂